.github/workflows/test_template.yml

```diff
@@ -43,6 +43,41 @@ jobs:
       - name: Install Dependencies
         run: mix deps.get
 
+      - name: Compile dependencies
+        run: mix compile
+
+  lint_codebase:
+    name: Linting
+    
+    needs: install_and_compile_dependencies
+    
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.head_ref }}
+
+      - name: Set up Elixir ${{ env.ELIXIR_VERSION }} and OTP ${{ env.OTP_VERSION }}
+        uses: erlef/setup-elixir@v1
+        with:
+          otp-version: ${{ env.OTP_VERSION }}
+          elixir-version: ${{ env.ELIXIR_VERSION }}
+
+      - name: Cache Elixir build
+        uses: actions/cache@v2
+        with:
+          path: |
+            _build
+            deps
+          key: ${{ runner.os }}-mix-${{ hashFiles('**/mix.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-mix-
+
+      - name: Install Dependencies
+        run: mix deps.get
```

Comment:
We can't skip that step as I remember @olivierobert thus the cached is there and there is no new dependency added but we still need to trigger the `mix deps.get` so Mix can understand that `mix deps.get` was done, without that, it will raise the error like `dependency doesn't install` 

I will make a demo and then upload it here.

Reply:
I have added a commit to test it.

1/ Commit

![image](https://user-images.githubusercontent.com/11751745/153796398-34b72b76-e09a-42eb-bb81-6c42f339a818.png)

2/ Result 

Install dependency is run on the first step - Install and Compile Dependencies 

![image](https://user-images.githubusercontent.com/11751745/153796458-12fd512f-b841-475e-a5ae-fa874c3a77b6.png)

On the second step (without the `ix deps.get` step above) it failed

![image](https://user-images.githubusercontent.com/11751745/153796556-e61c2f10-e23f-49da-9cda-279abe0d0819.png)

Reply:
All clear 👍